### PR TITLE
[pkg] implement NuGet ecosystem parser

### LINKED ISSUE #51: Scan nuget
Report on nuget package usage that has a vulnerability.  
- packages.config
- packagereference nodes in 
  - *.*proj 
  - *.props files

https://devblogs.microsoft.com/nuget/how-to-scan-nuget-packages-for-security-vulnerabilities/

### PR DESCRIPTION
**Pull request by shawnfunke**
Adds support to scan `packages.lock.json` files which are lock files
for the NuGet ecosystem.

Closes #51

### CODE REVIEW

**Review by G-Rath — APPROVED** [review]

**Comment by G-Rath** [comment]
Would you mind putting the "Closes #51" in the PR description? linking doesn't work with comments, and I can't edit PR descriptions :)

**Comment by inferno-chromium** [comment]
Thank you for your PR. The current committers are on vacation and we will review it early next year and then release as part of the the next minor release 1.0.x release.

File: pkg/lockfile/parse-nuget-lock.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"fmt"
+	"os"
+)
+
+type NuGetLockPackage struct {
+	Resolved string `json:"resolved"`
+}
+
+type NuGetLockfile struct {
+	Version      int                                    `json:"version"`
```
Comment (oliverchang) [inline pkg/lockfile/parse-nuget-lock.go:14]: Is there a reference to the spec that we can leave as a comment here?

Also curious if we should be checking the "version" here.

**Review by oliverchang — APPROVED** [review]
thank you very much for the contribution!

**Review by oliverchang — APPROVED** [review]
thanks!

**Pull request merged**